### PR TITLE
Rework toolbar design

### PR DESCRIPTION
**Pull request by julienamsellem**
### Purpose of this PR
Jira
https://jira.unity3d.com/browse/GVFXG-143
Design docs
https://docs.google.com/document/d/1oG7-tKzlEDqQCRnw0KVsRqldVUyTY5gBfVuF-5tPx2Q/edit
https://docs.google.com/document/d/1VS4LKjZmfx40kutvb39h3sIL6UCSM9m92IxuluxGTeI/edit#

Before
old-toolbar
After
new-toolbar

---
### Testing status
Tested on dark and light themes with 100% and 150% DPI screens

---
### Comments to reviewers
**Known bug**: when DPI is different than 100% the drop down popups can be position at a "not ideal" position.
The bug is most probably on UIToolkit side. I warned the [#dev-uitoolkit](https://unity.slack.com/archives/C3414V4UV/p1628498431112200) channel about it

### CODE REVIEW

File: com.unity.visualeffectgraph/Editor/GraphView/Views/VFXView.cs [hunk 1]
```diff
@@ -1451,6 +1389,15 @@ void OnSave()
             }
         }
 
+        internal void SaveAs(string newPath)
```
Comment (julienamsellem) [inline com.unity.visualeffectgraph/Editor/GraphView/Views/VFXView.cs:1392]: This method is not working yet, I'm still working on it

**Comment by VitaSkruibyte** [comment]
Dropping the WIP [test doc](https://docs.google.com/document/d/1kQJoc8r1FEsgYuoLllVdDC3v9PnjtMYfqStyxqEdFSo/edit?usp=sharing) for visibility

File: com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs [hunk 2]
```diff
@@ -106,6 +106,12 @@ void InternalLoadResource(VisualEffectResource resource)
             graphView.controller = VFXViewController.GetController(resource, true);
             graphView.UpdateGlobalSelection();
             graphView.FrameNewController();
+            graphView.UpdateIsSubgraph();
```
Comment (julienamsellem) [inline com.unity.visualeffectgraph/Editor/GraphView/VFXViewWindow.cs:109]: This piece of change is to support the new button in the toolbar to go back to main graph when a subgraph is opened

File: com.unity.visualeffectgraph/Editor/GraphView/Views/DropDownButtonBase.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
+using System;
```
Comment (julienamsellem) [inline com.unity.visualeffectgraph/Editor/GraphView/Views/DropDownButtonBase.cs:1]: This is a reusable class to add dropdowns (a button and an arrow button to open a popup).
It's used three times in this PR

File: com.unity.visualeffectgraph/Editor/GraphView/Views/DropDownButtonBase.cs [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+        private void OnPopupClosed()
+        {
+            m_CurrentPopup.Destroyed -= OnPopupClosed;
+            m_CurrentPopup.rootVisualElement.UnregisterCallback<KeyUpEvent>(OnKeyUp);
+            m_CurrentPopup = null;
+            m_PopupClosedTimestamp = DateTime.UtcNow;
+        }
+
+        private void OnKeyUp(KeyUpEvent evt)
```
Comment (julienamsellem) [inline com.unity.visualeffectgraph/Editor/GraphView/Views/DropDownButtonBase.cs:147]: These last three methods were added to handle keyboard navigation (either with Tab or arrow keys). This was requested by UX designers

**Review by VitaSkruibyte — APPROVED** [review]
Looking good! All the raised issues are fixed, couldn't find anything outstanding on the last QA pass. 🟢 

Thanks for the great PR, new toolbar looking very elegant in VFX window!

[Test Doc](https://docs.google.com/document/d/1kQJoc8r1FEsgYuoLllVdDC3v9PnjtMYfqStyxqEdFSo/edit)

**Pull request merged**